### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by Navijava**

### CODE REVIEW

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    @Test
+    void register_negativeAge_notOk() {
+        user.setAge(-26);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_UserNull_notOK() {
+        User user2 = new User();
```
Comment (DaniilOpryshko) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:67]:
```suggestion
        User user2 = null;
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        user.setAge(101);
+        assertThrows(RuntimeException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        user.setAge(-26);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
```
Comment (DaniilOpryshko) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:63]: You already have test that checks if age is undex 18

**Review by DaniilOpryshko — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java (hunk 2 above)
Comment (Navijava) [reply src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:63]: Yeap...

**Review by DaniilOpryshko — APPROVED** [review]
Good Job!

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        user.setAge(101);
+        assertThrows(RuntimeException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_UserNull_notOK() {
+        User user2 = null;
+        assertThrows(RuntimeException.class,
+                () -> registrationService.register(user2));
```
Comment (KostyaNikiforov) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:63]:
```suggestion
        assertThrows(RuntimeException.class,
                () -> registrationService.register(null));
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+    private StorageDaoImpl storageDao = new StorageDaoImpl();
+    private User user = new User();
+    private User userWithTheSameLogin1 = new User();
+    private User userWithTheSameLogin2 = new User();
```
Comment (KostyaNikiforov) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:16]: you need these variables only in one method, there is no need to leave its in a class level

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+    private StorageDaoImpl storageDao = new StorageDaoImpl();
+    private User user = new User();
```
Comment (KostyaNikiforov) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:14]: Its can be final

**Review by KostyaNikiforov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**